rover-apps/common/src/main.cpp

```diff
@@ -0,0 +1,90 @@
+#include "AppConfig.h"
+#include "Logger.h"
+#include "mbed.h"
+
+Thread periodic_10s_thread(osPriorityNormal1);
+Thread periodic_1s_thread(osPriorityNormal2);
+Thread periodic_100ms_thread(osPriorityNormal3);
+Thread periodic_10ms_thread(osPriorityNormal4);
+Thread periodic_1ms_thread(osPriorityNormal5);
+
+void periodic_10s(void) {
+  auto startTime = Kernel::Clock::now();
+  for (Module* module : gModules) {
+    module->periodic_10s();
+  }
+
+  auto nextStartTime = startTime + 10s;
+  if (Kernel::Clock::now() > nextStartTime) {
+    Utility::logger << "Reseting periodic 10s task timing as it failed to hit the deadline!\n";
+    nextStartTime = Kernel::Clock::now() + 10s;
```

Comment:
Is this to prevent this task from starving other threads? 

my gut thought was that it doesnt make sense to add extra time if it misses a deadline. See this example:
0s: period # 1 starts
11s: period # 1 ends (or rather hits this point and sees that 11 > 10)
21s: period # 2 starts

because of this logic, there is now a 21 second gap between period 1 and period 2 of a function thats supposed to run every 10 seconds.

Reply:
I think its better to just emit a warning about it and still use the original nextStartTime. In my mind, if we are consistently missing timings, then thats a whole other issue and we need to reassess the implementation of some of the periodic functions being called in the modules. If its a one-off situation(idk maybe some calculation took longer in some specific instance), then it makes sense to me to just warn the user and skip a sleep for once instead of intentionally delaying the next run even more (which is sort of making the problem worse)

Reply:
True, I was thinking if say the 1kHz task kept missing the deadline it would keep running and starve all the other tasks, but at that point we should be addressing the problem itself